hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2732,3 +2741,228 @@ func reconcileMachineConfigServerService(svc *corev1.Service) error {
 	svc.Spec.ClusterIP = corev1.ClusterIPNone
 	return nil
 }
+
+func (r *HostedClusterReconciler) reconcileMachineApprover(ctx context.Context, hcluster *hyperv1.HostedCluster, hcp *hyperv1.HostedControlPlane) error {
+	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(controlPlaneNamespace), controlPlaneNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane namespace: %w", err)
+	}
+
+	// Reconcile machine-approver role
+	role := machineapprover.Role(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
+		return reconcileMachineApproverRole(role)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role: %w", err)
+	}
+
+	// Reconcile machine-approver service account
+	sa := machineapprover.ServiceAccount(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, sa, NoopReconcile)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver service account: %w", err)
+	}
+
+	// Reconcile machine-approver role binding
+	rolebinding := machineapprover.RoleBinding(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, rolebinding, func() error {
+		return reconcileMachineApproverRoleBinding(rolebinding, role, sa)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role binding: %w", err)
+	}
+	config := machineapprover.ConfigMap(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, config, func() error {
+		return reconcileMachineApproverConfig(config)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver config: %w", err)
+	}
+
+	// The deployment depends on the kubeconfig being reported.
+	if hcp.Status.KubeConfig != nil {
+		// Resolve the kubeconfig secret for machine-approver
+		kubeconfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: hcp.Namespace,
+				Name:      "service-network-admin-kubeconfig",
+			},
+		}
```

Comment:
@enxebre  for hosted cluster's kubeconfig, I used `service-network-admin-kubeconfig`, which I copied from other components 😂 

it will be mounted onto machine-approver, and be referenced as `"--workload-cluster-kubeconfig=/etc/kubernetes/kubeconfig/kubeconfig"`

Reply:
ok make sense.